presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetHiveRecordCursor.java

```diff
@@ -741,15 +758,103 @@ public ParquetStructConverter(Type prestoType, String columnName, GroupType entr
                     prestoTypeParameters.size(),
                     fieldTypes.size());
 
-            this.rowType = prestoType;
-            this.fieldIndex = fieldIndex;
+            ImmutableList.Builder<BlockFieldConverter> converters = ImmutableList.builder();
+            for (int i = 0, n = prestoTypeParameters.size(); i < n; i++) {
+                parquet.schema.Type fieldType = fieldTypes.get(i);
+                converters.add(createConverter(prestoTypeParameters.get(i), columnName + "." + fieldType.getName(), fieldType, i, false));
+            }
+            return converters.build();
+        }
+
+        private List<BlockFieldConverter> createConvertersByName(String columnName, List<RowField> prestoTypeFields, List<parquet.schema.Type> fieldTypes)
```

Comment:
It's better to use createConvertersByName/createConvertersUseNames.
Changed prestoTypeFields to prestoRowFields. fieldTypes is fine.